sql/parser/eval.go

```diff
@@ -65,6 +68,21 @@ func (d DBool) Type() string {
 	return "bool"
 }
 
+// Compare implements the Datum interface.
+func (d DBool) Compare(other Datum) int {
+	v, ok := other.(DBool)
```

Comment:
I've added explanatory comments to the implementations. The expectation is that all the values in a column will either have a particular datum type or `dNull`. We shouldn't ever have to compare a `DInt` vs a `DString`. But now that I write that I'm not sure if it is true as expressions like `CASE` can return different types. Let me leave it as-is for now with a `TODO` in the `sortNode` code.